src/redux/data/dataSlice.js

```diff
@@ -0,0 +1,37 @@
+import { buildUrl } from '../../shared/buildUrl'
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
+
+const fetchData = createAsyncThunk('fetchData', (payload) => {
+  console.log(`fetching...`)
+  console.log(payload)
+  return fetch(buildUrl(payload.description, payload.location))
+    .then((response) => {
```

Comment:
meglio usare async/await ed evitare i then in generale se non in casi particolari